### PR TITLE
Use libscrypt from Hunter

### PR DESCRIPTION
**Pull request by chfast**
I fixed 2 tests, disabled 1.

The disabled test tests keyfile v2. I'm not sure what the expected result should be. I've found this commit adding the compatibility with v2: https://github.com/ethereum/cpp-ethereum/commit/8399161a8e9b4f0cf3a38e14f5c83e566aad21c2 but it does not include the test. I cannot find where the test was added.

Resolves https://github.com/ethereum/cpp-ethereum/issues/4815.

### CODE REVIEW

**Comment by axic** [comment]
```

The following tests FAILED:
	  2 - RandomCodeTests/rndStateTest (SEGFAULT)
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -155,6 +158,3 @@ if (TESTS)
     enable_testing()
     add_subdirectory(test)
 endif()
-
-# TODO: Split out json_spirit, libscrypt and sec256k1.
-add_subdirectory(utils)
```
Comment (axic) [inline CMakeLists.txt]: So its not even needed for `json_spirit`?

Comment (chfast) [reply CMakeLists.txt]: We use json_spirit by loading headers only. This is an old CMake script that was not used.

**Comment by chfast** [comment]
This test is failing from time to time only on CircleCI / macOS. To be disabled, because we cannot find the reason.

**Pull request merged**